packages/core/src/components/popover/popover.tsx

```diff
@@ -791,7 +786,7 @@ export class Popover<
 
     private getIsContentEmpty() {
         const { content } = this.props;
-        return content == null || (typeof content === "string" && content.trim() === "");
+        return Utils.isReactNodeEmpty(content) || Utils.isEmptyString(content);
```

Comment:
I don't think we should change this. If consumers want this behavior it's trivial for them to call `.trim`.

We could have done this from the beginning but on the off chance something relies on displaying an empty popover for empty space this doesn't feel worth even a 0.1% chance of a fire drill to address this.

Reply:
>  If consumers want this behavior it's trivial for them to call `.trim`

Existing code is already calling `trim`. This refactor is really just using existing functions to replace the same code that does the same thing.


```tsx
export function isEmptyString(val: any) {
    return typeof val === "string" && val.trim().length === 0;
}
```

It's literally the same code as previous. And then `isReactNodeEmpty` accounts for the left side of previous.

Reply:
But if you _are_ interested in changing the functionality, I do agree with your statement that it makes sense for it to be on the consumer to not pass empty content, not us to determine that.

Reply:
okay right, seeing

`typeof content === "string" && content.trim() === ""`

is functionally equal to

```
export function isEmptyString(val: any) {
    return typeof val === "string" && val.trim().length === 0;
}
```

Reply:
I think I was looking into both sides of this and got mixed up when writing a response

previously `[]`, or something like `["", "", undefined, null, []]` would render an empty content popover, now we'd consider this empty

Reply:
this seems less likely to be bad but this still feels like something we don't need to be messing with outside of a major version

Reply:
> previously `[]`, or something like `["", "", undefined, null, []]` would render an empty content popover, now we'd consider this empty

That's probably a good thing, think of this as a bug fix

Reply:
@evansjohnson  reverted